From patchwork Wed Jan 26 21:01:37 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Geis <pgwipeout@gmail.com>
X-Patchwork-Id: 12725790
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id CB3BEC433EF
	for <linux-rockchip@archiver.kernel.org>;
 Wed, 26 Jan 2022 21:01:53 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:Cc
	:To:From:Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:
	List-Owner; bh=iLzuhEIilL6AUzXnzvJMsSMjDCT36UMfBOhc2L2vKok=; b=0nhaHLITqCWD5q
	mdY3g67mEcnjOGTW6oKR+1p4KiXIUf/WLWKdGvB+NZaKQIBVhY2Qk0ZVBMJMPHyhWpXMzstF7uk+A
	0nf7zIKCIU8y3eX509wgRIqSg/iqSaTOgoamxJks/FeQsrAF/nmpT7BxhuUcPWR4pcVxqYOwFLNCx
	BSx+J/qeRTPMEnxt2HrAn8/+iNH5x2JsgyqwBHYwXNlPXexJyfFF5n7/6eB96gEmaRNBnADl5FPcN
	1cblQA7cJTYD7/7d5IB3ADOysJGs+apHNkPQUu1RZij/+rPNrYRFS7QSF26nqRLLEac+k0AB/Kuvs
	qAIeiDHYebkWLp8HRipg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nCpQJ-00DWiR-Go; Wed, 26 Jan 2022 21:01:47 +0000
Received: from mail-qk1-x72e.google.com ([2607:f8b0:4864:20::72e])
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1nCpQG-00DWi8-M8
 for linux-rockchip@lists.infradead.org; Wed, 26 Jan 2022 21:01:45 +0000
Received: by mail-qk1-x72e.google.com with SMTP id 71so785115qkf.4
 for <linux-rockchip@lists.infradead.org>;
 Wed, 26 Jan 2022 13:01:43 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20210112;
 h=from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=oCWuC818kxptSSDVBuEBTaU6eMFMBR0V0DEYlw49JmI=;
 b=WBd1S9ReDJVLCy600I/GwQ4fkCVbs4o8jCYoLM5Ga8rFrYobPBN3cVRi5KPUyVHAvX
 PAbMzcDUWZ86T1lShRkuJd+mg/440U77DNFBvFtwVXMMKRYpMs8FC9OvkTBznQagWYFK
 2Ys+mv4AsBola0xY8ohxKl61FA/3rCnzhp/9WU3Ypf02mSf3cvMaDDpKidl5tMp0XqDF
 yRr3lum6LzSkfpnVKWZgskut+D+gSMO0aVEkg+IluapNtRWrbabSAPg89TPRnH9xv7DC
 bcj9sb2wMRYBtOmmNbY63A1R0wEYudJMKK1ZpzNHRbrJN3DTRVWOj1pDKk+wX+QMddu8
 fmvQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20210112;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=oCWuC818kxptSSDVBuEBTaU6eMFMBR0V0DEYlw49JmI=;
 b=iJJ0NHUlWf/rm2N/PIxAZnsAq6FN0hWNmbF+fsL/lkOv46M7VvigqIKo2iAqfD1C4v
 vdiHL5iI/Y7IPcTa4yt6W8zEHlFK47crV8LHVlNocY/poqIvSt0NKJ30GZYpVkqtRsu2
 cchxIancKz2skwtMRxcVLPs8Lavi5kHEfdZ5W/3n+bj2Ael4GqdXJDL5pQh6hXsLX4/h
 JkzFxEWIlXYA+Q0bhKr/vbQ+wkdxTeBnczTw3nsorwKHSCld0iNOsx16KZvcuhKL3twt
 EKZ69u/U6LY+xFYDTYbBcqEWRg6hXCkOkCqsib5jWXU15PVU9R0HdFx4V3+Jd9q3o+D4
 KYcA==
X-Gm-Message-State: AOAM531eT0qjc9Km3CIsgf0pySUrXsJkDt/UDBNgw/NnW14YaFAw9pnE
 5G821T9+isD49/nSwvmQ3U8=
X-Google-Smtp-Source: 
 ABdhPJzQwVBWVM0Of40I6QipLyaGCsyPdlGpui0r8AIQ5RKUGPZM8lcMOC7JoZi5KtQ7luELeGDmhA==
X-Received: by 2002:a37:9d42:: with SMTP id g63mr459670qke.83.1643230903092;
 Wed, 26 Jan 2022 13:01:43 -0800 (PST)
Received: from master-x64.sparksnet (c-98-233-193-225.hsd1.dc.comcast.net.
 [98.233.193.225])
 by smtp.gmail.com with ESMTPSA id m4sm237427qkp.117.2022.01.26.13.01.42
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Wed, 26 Jan 2022 13:01:42 -0800 (PST)
From: Peter Geis <pgwipeout@gmail.com>
To: Andrzej Hajda <andrzej.hajda@intel.com>,
 Neil Armstrong <narmstrong@baylibre.com>,
 Robert Foss <robert.foss@linaro.org>,
 Laurent Pinchart <Laurent.pinchart@ideasonboard.com>,
 Jonas Karlman <jonas@kwiboo.se>, Jernej Skrabec <jernej.skrabec@gmail.com>,
 David Airlie <airlied@linux.ie>, Daniel Vetter <daniel@ffwll.ch>,
 Archit Taneja <architt@codeaurora.org>, Pierre-Hugues Husson <phh@phh.me>
Cc: Peter Geis <pgwipeout@gmail.com>, Sascha Hauer <s.hauer@pengutronix.de>,
 Robin Murphy <robin.murphy@arm.com>, linux-rockchip@lists.infradead.org,
	=?utf-8?q?Heiko_St=C3=BCbner?= <heiko@sntech.de>,
 dri-devel@lists.freedesktop.org, linux-kernel@vger.kernel.org
Subject: [PATCH v2] drm/bridge: synopsys/dw-hdmi: set cec clock rate
Date: Wed, 26 Jan 2022 16:01:37 -0500
Message-Id: <20220126210137.3065508-1-pgwipeout@gmail.com>
X-Mailer: git-send-email 2.25.1
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220126_130144_745607_19EBB673 
X-CRM114-Status: GOOD (  10.57  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

The hdmi-cec clock must be 32khz in order for cec to work correctly.
Ensure before enabling the clock we set it in order for the hardware to
work as expected.
Fixes hdmi-cec support on Rockchip devices.

Fixes: ebe32c3e282a ("drm/bridge: synopsys/dw-hdmi: Enable cec clock")

Signed-off-by: Peter Geis <pgwipeout@gmail.com>
---
Changelog:
v2:
- Set the clock rate before enabling the clock
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index 54d8fdad395f..65c16455b76a 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -48,6 +48,9 @@
 
 #define HDMI14_MAX_TMDSCLK	340000000
 
+/* HDMI CEC needs a clock rate of 32khz */
+#define HDMI_CEC_CLK_RATE	32768
+
 enum hdmi_datamap {
 	RGB444_8B = 0x01,
 	RGB444_10B = 0x03,
@@ -3341,6 +3344,10 @@ struct dw_hdmi *dw_hdmi_probe(struct platform_device *pdev,
 		hdmi->cec_clk = NULL;
 		goto err_iahb;
 	} else {
+		ret = clk_set_rate(hdmi->cec_clk, HDMI_CEC_CLK_RATE);
+		if (ret)
+			dev_warn(hdmi->dev, "Cannot set HDMI cec clock rate: %d\n", ret);
+
 		ret = clk_prepare_enable(hdmi->cec_clk);
 		if (ret) {
 			dev_err(hdmi->dev, "Cannot enable HDMI cec clock: %d\n",
